tests/handlers/aws/test_logstash_output.py

```diff
@@ -0,0 +1,288 @@
+        )
+
+        self.group_name = group_name
+        self.stream_name = stream_name
+
+        os.environ["AWS_DEFAULT_REGION"] = aws_default_region
+        os.environ["SQS_CONTINUE_URL"] = _sqs_create_queue(self.sqs_client, _class_based_id(self, suffix="-continuing"))
+        os.environ["SQS_REPLAY_URL"] = _sqs_create_queue(self.sqs_client, _class_based_id(self, suffix="-replay"))
+
+        self.mocks = {
+            "s3client": mock.patch("storage.S3Storage._s3_client", new=self.s3_client),
+            "cloudwatchclient": mock.patch("handlers.aws.utils.get_cloudwatch_logs_client", lambda: self.logs_client),
+            "sqsclient": mock.patch("handlers.aws.utils.get_sqs_client", lambda: self.sqs_client),
+            "sqsclient2": mock.patch("handlers.aws.handler.get_sqs_client", lambda: self.sqs_client),
+        }
+        for k, m in self.mocks.items():
+            m.start()
+
+    def tearDown(self) -> None:
+        self.localstack.stop()
+        self.logstash.stop()
+
+        for k, m in self.mocks.items():
+            m.stop()
+
+    def test_sent_messages(self) -> None:
+        os.environ["S3_CONFIG_FILE"] = _prepare_config_file(
+            self,
+            "config.yaml",
+            dict(CloudwatchLogStreamARN=self.cloudwatch_group_arn, LogstashURL=self.logstash.get_url()),
+            "folder/config.yaml",
+        )
+
+        event_cloudwatch_logs, event_ids_cloudwatch_logs = _logs_retrieve_event_from_cloudwatch_logs(
+            self.logs_client, group_name=self.group_name, stream_name=self.stream_name
+        )
+
+        ctx = ContextMock(TIMEOUT_15m)
+        handler(event_cloudwatch_logs, ctx)  # type: ignore
+
+        msgs = self.logstash.get_messages()
+        assert len(msgs) == 2
+
+        assert msgs[0]["message"] == self.fixtures["cw_log_1"].rstrip("\n")
+        assert msgs[1]["message"] == self.fixtures["cw_log_2"].rstrip("\n")
+
+    def test_failure_sending_messages(self) -> None:
+        os.environ["S3_CONFIG_FILE"] = _prepare_config_file(
+            self,
+            "config.yaml",
+            # NOTE: using a bogus URL to mimic sending failure
+            dict(CloudwatchLogStreamARN=self.cloudwatch_group_arn, LogstashURL="http://fake.url"),
+            "folder/config2.yaml",
+        )
+
+        event_cloudwatch_logs, event_ids_cloudwatch_logs = _logs_retrieve_event_from_cloudwatch_logs(
+            self.logs_client, group_name=self.group_name, stream_name=self.stream_name
+        )
+
+        ctx = ContextMock(TIMEOUT_15m)
+        handler(event_cloudwatch_logs, ctx)  # type: ignore
+
+        msgs = self.logstash.get_messages()
+        assert len(msgs) == 0
+
+        messages = _sqs_get_messages(self.sqs_client, os.environ["SQS_REPLAY_URL"])
+        assert len(messages) == 2
+
+        event1 = json_parser(messages[0]["Body"])
+        assert event1["event_payload"]["message"] == self.fixtures["cw_log_1"].rstrip("\n")
```

Comment:
I'd assert everything we expect in `event1["event_payload"]`, similarly like we do [here](https://github.com/elastic/elastic-serverless-forwarder/pull/210/files#diff-405288248598cf246486803f67aaf3389026f2e92321efdfa325efe6c42094daR268-R277)